src/higley_lab_to_nwb/benisty_2022/benisty_2022_spike2ttlinterface.py

```diff
@@ -0,0 +1,93 @@
+from typing import List
+import numpy as np
+from neo import io
+
+from neuroconv.datainterfaces.ecephys.baserecordingextractorinterface import BaseRecordingExtractorInterface
+from neuroconv.tools import get_package
+from neuroconv.utils import FilePathType
+from neuroconv.tools.signal_processing import get_rising_frames_from_ttl
+from spikeinterface.extractors import CedRecordingExtractor
+def _test_sonpy_installation() -> None:
+    get_package(
+        package_name="sonpy",
+        excluded_python_versions=["3.10", "3.11"],
+        excluded_platforms_and_python_versions=dict(darwin=dict(arm=["3.8", "3.9", "3.10", "3.11"])),
+    )
+def get_stream_ids_and_names(file_path: FilePathType) -> List[str]:
+    """Return a list of channel names as set in the recording extractor."""
+    r = io.CedIO(filename=file_path)
+    signal_channels = r.header["signal_channels"]
+    stream_ids = signal_channels["id"]
+    stream_names = signal_channels["name"]
+    return stream_ids, stream_names
+
+class Benisty2022Spike2TTLInterface(BaseRecordingExtractorInterface):
+    #TODO find a better name for the interface. It needs to be general for all type of signals not only TTL (e.g Wheel Motion) 
+    """
+    Data interface class for converting Spike2 synchronization signals from CED (Cambridge Electronic
+    Design) using the :py:class:`~spikeinterface.extractors.CedRecordingExtractor`."""
+
+    display_name = "Spike2 Recording"
+    # keywords = BaseRecordingExtractorInterface.keywords + ("CED",)
+    associated_suffixes = (".smr", ".smrx")
+    info = "Interface for Spike2 recording synchronization signals from CED (Cambridge Electronic Design)."
+
+    ExtractorName = "CedRecordingExtractor"
+
+    def __init__(self, file_path: FilePathType, stream_id: str = None, stream_name: str = None , verbose: bool = True, es_key: str = "ElectricalSeries"):
```

Comment:
Yeah, but I think I need to separate the interfaces then, because the wheel motion it's not a TTL signal. I implemented this interface form the actual TTL signal interface (#4) but I am not convinced this si the right way

Reply:
Either this or finding a more general way to describe all signals that are not neural signals

Reply:
Okay now I'm confused now because both #4 and this add this TTL interface 😄 Okay so can we consider wheel motion as analog signal then? You can then add a separate interface called `Spike2AnalogRecordingInterface`

Reply:
As discussed in the last ASAP project updates I will keep a unique interface for all signals coming from the same file.